### PR TITLE
feat: add unused files analyzer and its reporters

### PR DESCRIPTION
**Pull request by incendial**
### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix
[ ] New rule
[ ] Changes an existing rule
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[X] Other, please explain: new analyzer

### CODE REVIEW

File: lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/lint_console_reporter.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import 'dart:io';
```
Comment (dkrutskikh) [inline lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/lint_console_reporter.dart:1]: @incendial a bit strange, the file named `lint_console_reporter.dart`, with contained `UnusedFilesConsoleReporter` class

File: lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/json/unused_files_json_reporter.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    final encodedReport = json.encode({
+      'formatVersion': formatVersion,
+      'timestamp': reportTime.toString(),
+      'unusedFiles': records.map(_analysisRecordToJson).toList(),
+    });
+
+    output.write(encodedReport);
+  }
+
+  Map<String, Object> _analysisRecordToJson(UnusedFilesFileReport report) => {
```
Comment (dkrutskikh) [inline lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/json/unused_files_json_reporter.dart:38]: @incendial can you return `Map<String, String>` ?

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

File: lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/lint_console_reporter.dart (hunk 1 above)
Comment (incendial) [reply lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/console/lint_console_reporter.dart]: Fixed.

File: lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/json/unused_files_json_reporter.dart (hunk 2 above)
Comment (incendial) [reply lib/src/analyzers/unused_files_analyzer/reporters/reporters_list/json/unused_files_json_reporter.dart]: Fixed.

File: lib/src/analyzers/unused_files_analyzer/unused_files_analyzer.dart [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            .where((entity) => !isExcluded(
+                  relative(entity.path, from: rootFolder),
+                  config.globalExcludes,
+                ))
+            .map((entity) => entity.path))
+        .toList();
+
+    final unusedFiles = filePaths.toSet();
+
+    for (final filePath in filePaths) {
```
Comment (dkrutskikh) [inline lib/src/analyzers/unused_files_analyzer/unused_files_analyzer.dart:57]: @incendial please rewrite like this https://github.com/dart-code-checker/dart-code-metrics/blob/master/lib/src/analyzers/lint_analyzer/lint_analyzer.dart#L98

**Review by dkrutskikh — CHANGES_REQUESTED** [review]

**Review by dkrutskikh — APPROVED** [review]

**Pull request merged**